service/src/main/java/bio/terra/workspace/service/resource/referenced/cloud/any/datareposnapshot/ReferencedDataRepoSnapshotResource.java

```diff
@@ -164,6 +165,25 @@ public boolean checkAccess(FlightBeanBag context, AuthenticatedUserRequest userR
     return dataRepoService.snapshotReadable(instanceName, snapshotId, userRequest);
   }
 
+  @Override
+  public WsmResource buildReferencedClone(
+      UUID destinationWorkspaceUuid,
+      @org.jetbrains.annotations.Nullable UUID destinationResourceId,
+      @org.jetbrains.annotations.Nullable UUID destinationFolderId,
+      @org.jetbrains.annotations.Nullable String name,
+      @org.jetbrains.annotations.Nullable String description) {
+    ReferencedDataRepoSnapshotResource.Builder resultBuilder =
```

Comment:
176-184 look the same for all resource types, should this go in a common place/super class?

Reply:
It is not. Someone wrote these `toBuilder` classes that hide the resource-specific settings.
I am not a fan of that approach. Look here: https://github.com/DataBiosphere/terra-workspace-manager/blob/63f703fa396948e4d79cef44cf93aac9bcdcded0/service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gcsbucket/ControlledGcsBucketResource.java#L151

That is my preference and it does not require a `toBuilder`. It constructs the new object.